src/twisted/conch/ssh/transport.py

```diff
@@ -873,11 +897,16 @@ def ssh_KEXINIT(self, packet):
         )
         self.outgoingCompressionType = ffs(client[6], server[6])
         self.incomingCompressionType = ffs(client[7], server[7])
-        if None in (
-            self.kexAlg,
-            self.keyAlg,
-            self.outgoingCompressionType,
-            self.incomingCompressionType,
+        if (
+            None
+            in (
+                self.kexAlg,
+                self.keyAlg,
+                self.outgoingCompressionType,
+                self.incomingCompressionType,
+            )
+            # RFC 8308, section 2.2
+            or self.kexAlg in (b"ext-info-c", b"ext-info-s")
```

Comment:
Just more comments so that you don't have to open the RFC and see what 2.2 is :)
```suggestion
            # RFC 8308, section 2.2
            # Most probably a broken implementation on the remote peer,
            # as the extension markers should not be returned as the negotiated algorithm.
            or self.kexAlg in (b"ext-info-c", b"ext-info-s")
```